src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -339,6 +340,12 @@ private void OnCollectionReset()
         {
             OnCollectionChanged(EventArgsCache.ResetCollectionChanged);
         }
+
+        private void MonitorEnter() => ++_monitorBusyCount;
+
+        private void MonitorExit() => --_monitorBusyCount;
```

Comment:
One thought:
What if you were to change MonitorExitDisposable to be a struct?  Then instead of these extra helpers and the custom try/catch in OnCollectionChanged, you could just use:
```C#
using (new MonitorExitDisposable(this))
{
    ...
}
```
?